Sources/QuaternionModule/Transformation.swift

```diff
@@ -0,0 +1,415 @@
+  /// in all lanes.
+  /// - If the rotation angle is zero, the rotation vector is `nan`
+  /// in all lanes.
+  ///
+  /// See also:
+  /// -
+  /// - `.angle`
+  /// - `.axis`
+  /// - `.angleAxis`
+  /// - `init(angle:axis:)`
+  /// - `init(length:angle:axis)`
+  /// - `init(rotation:)`
+  ///
+  /// [rotvector]: https://en.wikipedia.org/wiki/Axis–angle_representation#Rotation_vector
+  @_transparent
+  public var rotationVector: SIMD3<RealType> {
+    axis * angle
+  }
+
+  /// The [polar decomposition][wiki].
+  ///
+  /// Returns the length of this quaternion, phase in radians of range *[0, π]*
+  /// and the rotation axis as SIMD3 vector of unit length.
+  ///
+  /// Edge cases:
+  /// -
+  /// - If the quaternion is zero, length is `.zero` and angle and axis
+  /// are `nan`.
+  /// - If the quaternion is non-finite, length is `.infinity` and angle and
+  /// axis are `nan`.
+  /// - For any length other than `.zero` or `.infinity`, if angle is zero, axis
+  /// is `nan`.
+  ///
+  /// See also:
+  /// -
+  /// - `.angle`
+  /// - `.axis`
+  /// - `.angleAxis`
+  /// - `.rotationVector`
+  /// - `init(angle:axis:)`
+  /// - `init(length:angle:axis)`
+  /// - `init(rotation:)`
+  ///
+  /// [wiki]: https://en.wikipedia.org/wiki/Polar_decomposition#Quaternion_polar_decomposition
+  public var polar: (length: RealType, phase: RealType, axis: SIMD3<RealType>) {
+    (length, halfAngle, axis)
+  }
+
+  /// Creates a unit quaternion specified with [Angle-Axis][wiki] values.
+  ///
+  /// Angle-Axis is a representation of a 3D rotation using two different
+  /// quantities: an angle describing the magnitude of rotation, and a vector
+  /// of unit length indicating the axis direction to rotate along.
+  ///
+  /// This initializer reads given `angle` and `axis` values and creates a
+  /// quaternion of equal rotation properties using the following equation:
+  ///
+  ///     Q = (cos(angle/2), axis * sin(angle/2))
+  ///
+  /// Given `axis` gets normalized if it is not of unit length.
+  ///
+  /// The final quaternion is of unit length.
+  ///
+  /// Edge cases:
+  /// -
+  /// - For any `θ`, even `.infinity` or `.nan`:
+  ///   ```
+  ///   Quaternion(angle: θ, axis: .zero) == .zero
+  ///   ```
+  /// - For any `θ`, even `.infinity` or `.nan`:
+  ///   ```
+  ///   Quaternion(angle: θ, axis: .infinity) == .ininfity
+  ///   ```
+  /// - Otherwise, `θ` must be finite, or a precondition failure occurs.
+  ///
+  /// See also:
+  /// -
+  /// - `.angle`
+  /// - `.axis`
+  /// - `.angleAxis`
+  /// - `.rotationVector`
+  /// - `.polar`
+  /// - `init(rotation:)`
+  /// - `init(length:angle:axis)`
+  ///
+  /// - Parameter angle: The rotation angle about the rotation axis in radians
+  /// - Parameter axis: The rotation axis
+  ///
+  /// [wiki]: https://en.wikipedia.org/wiki/Quaternions_and_spatial_rotation#Recovering_the_axis-angle_representation
+  @inlinable
+  public init(angle: RealType, axis: SIMD3<RealType>) {
```

Comment:
Should this take an explicit `length` parameter (with default value 1), so that it's dual to `polar`?

Reply:
> * Should this take an explicit `length` parameter (with default value 1), so that it's dual to `polar`?

Thats probably a great idea. Also, I was a bit worried that people might mistakenly use the polar initializer for a scaled Quaternion from an Angle/Axis representation, when the really should not. This would help separate them both clearly and hopefully guide people to the correct initializer. I'll add it, thank you!

> * Instead of normalizing, we could require that `axis` is normal (with a debug assert check?)

I think, at some point, it would be nice to add a construct that guarantees unity of an underlying AlgebraicField type (something like `struct Unit<T: AlgebraicField> { var normalized: T? }`) and only those are assumed to have unit length, while others are not. However, `axis` is arguably always of unit length (or should be), and as far as I can tell, most other libraries just assume it is of unit length, so I am a bit back and forth on this. If we go with a debug assert check, might I ask, what would be a reasonable tolerance for the length of the axis?

Reply:
I implemented both of the suggested changes (Thanks!).
The initializer now features an additional length parameter (defaults to 1, if omitted) and the axis is assumed to be of unit length – instead of normalizing it – or it will raise an assertion failure.

However, there are now two initializer available that are almost equal, namely:

- `init(length: RealType = 1, angle: RealType, axis: SIMD3<RealType>)` for Angle-Axis
- `init(length: RealType, phase: RealType, axis: SIMD3<RealType>)` for Polar decomposition

and I am uncertain if this acceptable/fine.